bc_obps/reporting/tests/service/test_compliance_service/test_compliance_service_class.py

```diff
@@ -63,3 +64,21 @@ def test_new_entrant(self):
         assert result.emissions_limit == 0
         assert result.excess_emissions == 0
         assert result.credited_emissions == 0
+
+    def test_saves_compliance_data(self):
+        build_data = ComplianceTestInfrastructure.build()
+        result = ComplianceService.get_calculated_compliance_data(build_data.report_version_1.id)
+        ComplianceService.save_compliance_data(build_data.report_version_1.id)
+        report_compliance_summary_record = ReportComplianceSummary.objects.get(
+            report_version_id=build_data.report_version_1.id
+        )
+        report_compliance_product_records = ReportComplianceSummaryProduct.objects.filter(
+            report_compliance_summary_id=report_compliance_summary_record.id
+        )
+
+        assert (
+            report_compliance_summary_record.emissions_attributable_for_compliance
+            == result.emissions_attributable_for_compliance
+        )
+        for idx, p in enumerate(report_compliance_product_records):
+            assert p.annual_production == result.products[idx].annual_production
```

Comment:
This is only testing the `create` case, which is why it worked with the `update_and_create` bug above, we might want to also test the update case